pkg/kp/pcstore/consul_store.go

```diff
@@ -143,6 +159,42 @@ func (s *consulStore) pcPath(pcID fields.ID) (string, error) {
 	return path.Join(podClusterTree, pcID.String()), nil
 }
 
+func (s *consulStore) lockForUpdateCreation(podID types.PodID,
+	availabilityZone fields.AvailabilityZone,
+	clusterName fields.ClusterName,
+	session Session) (consulutil.Unlocker, error) {
+	return session.Lock(pcCreateLockPath(podID, availabilityZone, clusterName))
+}
+
+func pcCreateLockPath(podID types.PodID,
+	availabilityZone fields.AvailabilityZone,
+	clusterName fields.ClusterName) string {
+	return path.Join(kp.LOCK_TREE, podID.String(), availabilityZone.String(), clusterName.String())
+}
+
+func (s *consulStore) FindWhereLabeled(podID types.PodID,
+	availabilityZone fields.AvailabilityZone,
+	clusterName fields.ClusterName) ([]fields.PodCluster, error) {
+
+	sel := klabels.Everything().
```

Comment:
we should probably put the label set in an accessible place and then use `labelSet.AsSelector()`